optuna/storages/_cached_storage.py

```diff
@@ -344,19 +349,21 @@ def get_all_trials(
             trials = list(sorted(trials.values(), key=lambda t: t.number))
             return copy.deepcopy(trials) if deepcopy else trials
 
-    def read_trials_from_remote_storage(self, study_id: int) -> None:
+    def read_trials_from_remote_storage(self, study_id: int, sync_owned_trials: bool) -> None:
```

Comment:
Did you consider to implement two separate methods instead of having a boolean parameter?